mep/common/utils.py

```diff
@@ -63,45 +63,59 @@ def alpha_pagelabels(paginator, objects, attr_meth):
     for i in range(paginator.num_pages):
         page = paginator.page(i + 1)  # paginator page is 1-based
         # get objects at start & end of each page (paginator index also 1-based)
-        labels.append(attr_meth(objects[page.start_index() - 1]))
+        page_label = attr_meth(objects[page.start_index() - 1])
+        if max_chars:
+            page_label = page_label[:max_chars]
+        labels.append(page_label)
         # don't go beyond the end of the actual number of objects
         end_index = min(page.end_index() - 1, paginator.count)
         # add end label only if not the same as first (e.g., page of a single item)
         if page.start_index() - 1 != end_index:
-            labels.append(attr_meth(objects[end_index]))
-
-    # abbreviate labels so they are as short as possible but distinct from
-    # preceding and following labels
-    abbreviated_labels = []
-    # iterate over all labels and generate the shortest distinct version
-    for i, label in enumerate(labels):
-        next_label = labels[i + 1] if i + 1 < len(labels) else ''
-        prev_label = labels[i - 1] if i > 0 else ''
-
-        # In the rare case that two labels are *exactly* the same,
-        # just add the full label.
-        if label in (next_label, prev_label):
-            abbreviated_labels.append(label)
-            continue
-
-        # iterate through the label one letter at a time
-        # to determine the minimum length needed to differentiate
-        # start with one letter, go up to full length of the label if necessary
-        abbr = ''
-        for j, char in enumerate(label, 1):
-            abbr += char   # or abbr = label[:j]
-            # if abbreviation is different from neighboring labels
-            # at the current length, then use it
-            # - compare case-insensitively
-            # NOTE: should possibly ignore trailing punctuation here...
-            # possible to get "Gray" / "Gray,"
-            if abbr.lower() not in (next_label[:j].lower(),
-                                    prev_label[:j].lower()):
-                break
-            # if we don't break before the loop finishes, use
-            # the whole label
-
-        abbreviated_labels.append(abbr)
+            end_label = attr_meth(objects[end_index])
+            if max_chars:
+                end_label = end_label[:max_chars]
+            labels.append(end_label)
+
+    if max_chars:
+        abbreviated_labels = labels
+
+    else:
+        # abbreviate labels so they are as short as possible but distinct from
```

Comment:
codefactor flagged this as "complex method", presumably due to length...i think we could maybe split this portion (creating abbreviated labels) into another function, but up to you. if it seems super easy, maybe worth it.

Reply:
Was just about to ask that - we're not actually using the smart abbreviation behavior anymore, since we're truncating at a specific set of characters. So I could just remove it, which I imagine would simplify the method quite a bit. (I kind of hate to remove it because I think it's clever, but I also know that's silly.)

Reply:
Isn't this the only project we're used it on? I know you spent some time digging up the old code that powered this and adapting it, so it seems a shame to not have it stored anywhere (except git history, I guess). I'm fine with keeping the smart abbreviation here as its own method, maybe with a note about how it's not currently in use. As long as the `if/else` statement is still there, codefactor shouldn't flag it as dead.